File: leetcode/easy/time-needed-to-buy-tickets.py
# https://leetcode.com/problems/time-needed-to-buy-tickets/

class Solution:
    def timeRequiredToBuy(self, t: List[int], k: int) -> int:
        ans = 0
        for i in range(len(t)):
            if i <= k:
                ans += min(t[i], t[k])
            if i > k:
                ans += min(t[k]-1, t[i])
        return ans

File: leetcode/easy/time-needed-to-buy-tickets.ts
// https://leetcode.com/problems/time-needed-to-buy-tickets/

// MINE: brute force
/*
function timeRequiredToBuy(tickets: number[], k: number): number {
    let ans = 0
    while(true) {
        for(let i=0; i<tickets.length; ++i) {
            if(tickets[i] === 0) continue
            --tickets[i]
            ++ans
            if(tickets[k] === 0) return ans
        }
    }
    return -1
};
*/

// ANSWER: one pass
function timeRequiredToBuy(t: number[], k: number): number {
  let ans = 0
  for (let i = 0; i < t.length; ++i) {
    if (i <= k) ans += Math.min(t[i], t[k])
    if (i > k) ans += Math.min(t[k] - 1, t[i])
  }
  return ans
}
